src/carbonio-files-ui-common/views/components/NodeListItem.tsx

```diff
@@ -403,50 +426,153 @@ const NodeListItemComponent: React.VFC<NodeListItemProps> = ({
 		setIsContextualMenuActive(false);
 	}, []);
 
-	const preventTextSelection = useCallback<React.MouseEventHandler>((e: React.MouseEvent): void => {
-		if (e.detail > 1) {
-			e.preventDefault();
+	const [dropzoneEnabled, setDropzoneEnabled] = useState(isFolder(node));
+
+	const dragMoveHandler = useCallback(() => {
+		const draggedNodes = draggedItemsVar();
+		if (draggedNodes && draggedNodes.length > 0 && canBeMoveDestination(node, draggedNodes, me)) {
+			navigationTimerRef.current = setTimeout(() => {
+				navigateTo(node.id);
+			}, TIMERS.DRAG_NAVIGATION_TRIGGER);
+			return true;
 		}
+		return false;
+	}, [me, navigateTo, node]);
+
+	const dragUploadHandler = useCallback(() => {
+		navigationTimerRef.current = setTimeout(() => {
+			navigateTo(node.id);
+		}, TIMERS.DRAG_NAVIGATION_TRIGGER);
+		return canUploadFile(node);
+	}, [navigateTo, node]);
+
+	const dragEnterHandler = useCallback(
+		(event) => {
+			// check if node is a valid destination for write inside action
+			setDropzoneEnabled((prevState) => {
+				navigationTimerRef.current && clearTimeout(navigationTimerRef.current);
+				if (prevState && event.dataTransfer.types.includes(DRAG_TYPES.move)) {
+					return dragMoveHandler();
+				}
+				if (event.dataTransfer.types.includes(DRAG_TYPES.upload)) {
+					return dragUploadHandler();
+				}
+				return false;
+			});
+		},
+		[dragMoveHandler, dragUploadHandler]
+	);
+	const dragLeaveHandler = useCallback(() => {
+		navigationTimerRef.current && clearTimeout(navigationTimerRef.current);
 	}, []);
 
-	return (
-		<Container data-testid={id}>
-			<ContextualMenu
-				disabled={
-					(disabled || isSelectionModeActive || compact) &&
-					selectionContextualMenuActionsItems === undefined
+	const moveNodesAction = useCallback(
+		(event) => {
+			const movingNodes = JSON.parse(event.dataTransfer.getData(DRAG_TYPES.move) || '{}');
+			if (movingNodes && isFolder(node)) {
+				moveNodesMutation(node, ...movingNodes).then(() => {
+					selectionModeVar(false);
```

Comment:
Not a fan, we used exitSelectionMode generally, we could probably improve the hook or keep passing it down as prop for now